projects/ngx-datatable/src/lib/components/datatable.component.ts

```diff
@@ -879,13 +879,17 @@ export class DatatableComponent implements OnInit, DoCheck, AfterViewInit {
     }
 
     this.offset = offset;
-
-    this.page.emit({
-      count: this.count,
-      pageSize: this.pageSize,
-      limit: this.limit,
-      offset: this.offset
-    });
+    // Re-assign to trigger offset calc in getter, otherwise values will differ
+    offset = this.offset;
+
+    if (!isNaN(offset)) {
```

Comment:
@sandrafreihofer that's a rather weird flow here and overwriting a parameter is also quite an anti-pattern:

```ts
this.offset = offset;
offset = this.offset;
```

You could prevent setting invalid values as well as having a simpler flow by using:

```ts
  onBodyPage({ offset }: any): void {
    // Avoid pagination caming from body events like scroll when the table
    // has no virtualization and the external paging is enable.
    // This means, let's the developer handle pagination by my him(her) self
    if (this.externalPaging && !this.virtualization || isNaN(offset)) {
      return;
    }

    this.offset = offset;

    this.page.emit({
      count: this.count,
      pageSize: this.pageSize,
      limit: this.limit,
      offset: this.offset
    });
  }
```

Reply:
nope, the `offset` that is passed is fine (it's 0). The problem is that `this.offset` is a getter and a setter, the setter returning the `NaN` during initialization. So maybe instead rename:

```ts
const localOffset = this.offset;
if (!isNaN(localOffset)) {
 ...
}
```

Reply:
In general, I'd like to replace the getters and setters with the Angular life cycle hooks either way, but I don't know if that's what we want to do, since any back ports to the original repo would be a bit more difficult. But I think this is another topic and is not important for this PR.

Reply:
using `this.offset` directly is good enough